### PR TITLE
Fixes related to nightly and alloy

### PR DESCRIPTION
**Pull request by LindaGuiga**
This PR introduces two fixes:
- Latest nightly made `env::set_var` unsafe, so this PR introduces an `unsafe` block when calling the method.
- It seems that the `alloy` library changed the second argument in `get_block` from a `bool` to a `BlockTransactionKind` (after updating rust, I got an error message complaining about the type of that argument), so this PR changes the `const` accordingly.

### CODE REVIEW

File: rpc/src/lib.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ enum ZeroTrace {
 /// - Just the hash.
 ///
 /// We only need the latter.
-const BLOCK_WITHOUT_FULL_TRANSACTIONS: bool = false;
+const BLOCK_WITHOUT_FULL_TRANSACTIONS: BlockTransactionsKind = BlockTransactionsKind::Hashes;
```
Comment (Nashtare) [inline rpc/src/lib.rs]: I'd rename the constant at this point, it's not really consistent with the type anymore.

**Review by Nashtare — COMMENTED** [review]
Note that the type change from `alloy` isn't due to nightly but to a bump in the deps. We should probably pin a specific commit for version control, so that we only upgrade at will, at least until the crates get published.

File: rpc/src/lib.rs (hunk 1 above)
Comment (LindaGuiga) [reply rpc/src/lib.rs]: Would something like `BLOCK_TRANSACTIONS_HASHES_ONLY` work?

**Comment by atanmarko** [comment]
@Nashtare @LindaGuiga Regression tested with Jerigon, seems all good.

**Review by atanmarko — APPROVED** [review]

**Review by muursh — APPROVED** [review]

**Comment by atanmarko** [comment]
Note: `#[allow(unused_unsafe)]` was added in order not to break `clippy` before latest nightly(`1.81.0-nightly`). We will remove it when it becomes redundant.

File: rpc/src/lib.rs (hunk 1 above)
Comment (frisitano) [reply rpc/src/lib.rs]: Does it make sense to have a constant anymore? I think using the enum itself reads easier.

**Comment by Nashtare** [comment]
Note that the PR has been merged without addressing this:
> Note that the type change from alloy isn't due to nightly but to a bump in the deps. We should probably pin a specific commit for version control

which is what caused the build to break in the first place.

**Comment by atanmarko** [comment]
@Nashtare My bad. I'll open another ticket to fix alloy versions. I could do that after we test and merge  https://github.com/0xPolygonZero/zero-bin/pull/81 ?

**Comment by Nashtare** [comment]
Yeah sure.

**Pull request merged**